### DISCUSSION TITLE
Is it possible to make SampleList public?

### DISCUSSION THREAD

**Discussion by garar** [jenetics/jenetics#870] [Ideas]
Hi!

There are two classes implementing `Sampling` interface in jenetics prog: `SampleBuffer` and `SampleList`. I would like to use `SampleList` but it is not public? Would it be possible to make it public? Should I prepare a PR or do you want to keep `SampleList` non-public?

`SampleBuffer` is public but it is much more complicated than my case requires.

Thanks!

**Comment by jenetics** [comment]
Do you want to extend from the `SampleList`? The `Sampling` interface has a factory method, which allows you to create a `Sampling` instance from a list of `Sample` points: [`<T> Sampling<T> of(final List<? extends Sample<T>> samples)`](https://www.javadoc.io/static/io.jenetics/jenetics.prog/7.1.3/io.jenetics.prog/io/jenetics/prog/regression/Sampling.html#of(java.util.List)).

**Reply by garar** [reply]
Hi! I didn't want to extend `SampleList`, just create a new instance. Your idea works great, sorry I missed it.

I can't seem to mark my question as answered.
